data/changelog/ppxlib/2024-07-25-ppxlib-0.33.0.md

```diff
@@ -0,0 +1,111 @@
+---
+title: Ppxlib 0.33.0
+tags: [ppxlib, platform]
+changelog: |
+  - Fix a bug where `Code_path.main_module_name` would not properly remove
+  extensions from the filename and therefore return an invalid module name.
+  (#512, @NathanReb)
+
+  - Add `-unused-type-warnings` flag to the driver to allow users to disable
+  only the generation of warning 34 silencing structure items when using
+  `[@@deriving ...]` on type declarations. (#511, @mbarbin, @NathanReb)
+
+  - Make `-unused-code-warnings` flag to the driver also controls the generation
+  of warning 34 silencing structure items when using `[@@deriving ...]` on type
+  declarations. (#510, @mbarbin, @NathanReb)
```

Comment:
```suggestion
  - Make `-unused-code-warnings` flag used with the driver also control the generation
  of warning 34, silencing structure items when using `[@@deriving ...]` on type
  declarations. (#510, @mbarbin, @NathanReb)
```

Although "Add `-unused...` flag to the driver" makes sense (Add...to), "Make `-unused...` flag to the driver" doesn't: (Make...to) 

Are you making `this code` work with the driver? or when "used" with the driver?

Also, for both li 9-11 and 13-15, does "...allow users to disable only the Warning 34 generation, silencing structure items when using..." In other words:
1. Is Warning 34 a specific warning where a capital letter would clarify that?
2. Does this generation silences structure items?
3. ...or does it generate "Warning 34 silencing structure items" ... when it's "using ... on type declarations"?

Reply:
Hmm you're write I think I copied "Add flag x to the driver that controls" from the item above and failed to properly edit it.

Let me give you a bit of context so we can figure out how to properly phrase this.

The flag existed before this release, we just modified it's behaviour to control other features.

We can add a capital 'w' as it is indeed a specific compiler warning, identified by its number.

It's poorly phrased and a bit long but we could say that the flag controls the generation of extra structure items that are emitted solely to silence (i.e. to locally disable) Warning 34.

I think my original intent was the following:
```md
- Make the `-unused-code-warnings` driver flag also control the generation of Warning 34 silencing structure
items when ...
```

Does that look good enough?

Would you prefer a simpler phrasing, maybe something along those lines:
```md
- Make the `-unused-code-warnings` driver flag also control Warning 34 silencing for type declarations
  with `[@@deriving ...]` attached.
```

Reply:
Oooh yes! The second one is great! Perfect!

"- Make the `-unused-code-warnings` driver flag also control Warning 34 silencing for type declarations with `[@@deriving ...]` attached."